### PR TITLE
Feature/BCSCP_18: demo app and terraform config

### PR DESCRIPTION
**Pull request by roy-sagar11**
## Summary
Added demo application and terraform deployment config

## Changes

- Added dot net demo application with keycloak login
- Added 2 applications in docker-compose for the demo app
- Added production dockerfile for demo app
- Added terraform configs to deploy 2 seperate instance of the demo app
- Added github workflow for terraform plan and deployment of the demo app

### CODE REVIEW

File: .github/workflows/demo-app-build-and-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        run: |
+          echo "${{secrets.KUBE_CONFIG}}" > ./terraform/demo-app/kubeconfig
+      - name: set kube-config
+        uses: azure/k8s-set-context@v2
+        with:
+          method: kubeconfig
+          kubeconfig: ${{ secrets.KUBE_CONFIG }}
+          context: ${{ secrets.KUBE_CONTEXT }}
+
+      - name: terraform destroy
```
Comment (simensma-fresh) [inline .github/workflows/demo-app-build-and-deploy.yml]: Do we need this `destroy` action? Seeing as we're passing along the Docker image version, it *should* redeploy it (hopefully)

**Review by simensma-fresh — CHANGES_REQUESTED** [review]

File: .github/workflows/demo-app-build-and-deploy.yml (hunk 1 above)
Comment (roy-sagar11) [reply .github/workflows/demo-app-build-and-deploy.yml]: oh! yeah sorry i forgot to update that. initially i added it for the latest tag. will update in a sec

**Review by simensma-fresh — APPROVED** [review]
Looks great!

**Review by iman-jamali-fw — APPROVED** [review]
One of the largest pr ever :)
Looks good to me

**Pull request merged**